drivers/ethernet/eth_stm32_hal.c

```diff
@@ -70,7 +70,7 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 #define __eth_stm32_desc __dtcm_noinit_section
 #define __eth_stm32_buf  __dtcm_noinit_section
 #elif defined(CONFIG_SOC_SERIES_STM32H7X) && \
-		DT_NODE_HAS_STATUS(DT_NODELABEL(sram3), okay)
+		DT_NODE_HAS_STATUS(DT_ALIAS(sramnocache), okay)
```

Comment:
This change means that only `STM32H7X` boards that define `sramnocache` will work with this driver now.
As this is more than just `nucleo_h745zi_q`, this will break the driver for many boards without further board updates.

Reply:
No. This is not. The other boards will still work. Please take a look on on first `#if` and last `#else`. There are many cases upon `sramnocache` already covered. 
There maybe some documentation effort for stm32 needed to advice users to use `sramnocache`.

Reply:
Ok, they will still work, but you are changing the behavior for every `STM32H7X` board apart from `nucleo_h745zi_q`. This is not necessarily wrong, but it is not explained anywhere in the commits as to why this is a good idea / why the previous way was bad.

Reply:
This is a good point. I would say, it would be nice to add some documentation to the whole he series. But there is no docs for it, as I know. Maybe it can be added to .yaml or the soc? @erwango?

Reply:
The point is not that it is not documented in the code, but that you have provided no justification for the change in the commit messages.